src/Sylius/Component/Addressing/Model/Country.php

```diff
@@ -131,7 +131,6 @@ public function removeProvince(ProvinceInterface $province)
     {
         if ($this->hasProvince($province)) {
             $this->provinces->removeElement($province);
-            $province->setCountry(null);
```

Comment:
I will split this PR into two, cause it would became messy as some relations needs to be adjusted too.